letsencrypt/scripts/main.py

```diff
@@ -9,61 +9,100 @@
+
+    add("-d", "--domains", metavar="DOMAIN", nargs="+")
+    add("-s", "--acme-server", "--server", default="letsencrypt-demo.org:443",
+        help="CA hostname (and optionally :port). The server certificate must "
+             "be trusted in order to avoid further modifications to the "
+             "client.")
+
+    add("-p", "--privkey", type=read_file,
+        help="Path to the private key file for certificate generation.")
+    add("-B", "--rsa-key-size", "--keysize", type=int, default=2048,
+        metavar="N", help="RSA key shall be sized N bits.")
+
+    add("-k", "--revoke", action="store_true", help="Revoke a certificate.")
+    add("-b", "--rollback", type=int, default=0, metavar="N",
+        help="Revert configuration N number of checkpoints.")
+    add("-v", "--view-config-changes", action="store_true",
+        help="View checkpoints and associated configuration changes.")
+    add("-r", "--redirect", action="store_true",
+        help="Automatically redirect all HTTP traffic to HTTPS for the newly "
+             "authenticated vhost.")
+
+    add("-e", "--agree-tos", dest="eula", action="store_true",
+        help="Skip the end user license agreement screen.")
+    add("-t", "--text", dest="use_curses", action="store_false",
+        help="Use the text output instead of the curses UI.")
+    add("--test", action="store_true", help="Run in test mode.")
+
+    # TODO: trailing slashes might be important! check and remove
+    add("--config-dir", default="/etc/letsencrypt/",
+        help="Configuration directory.")
+    add("--work-dir", default="/var/lib/letsencrypt/",
+        help="Working directory.")
+    add("--backup-dir", default="/var/lib/letsencrypt/backups/",
+        help="Configuration backups directory.")
+    add("--temp-checkpoint-dir",
+        default="/var/lib/letsencrypt/temp_checkpoint/",
+        help="Temporary checkpoint directory.")
+    add("--in-progress-dir",
+        default="/var/lib/letsencrypt/backups/IN_PROGRESS/",
+        help="Directory used before a permanent checkpoint is finalized")
+    add("--cert-key-backup", default="/var/lib/letsencrypt/keys-certs/",
+        help="Directory where all certificates and keys are stored. "
+             "Used for easy revocation.")
+    add("--rev-tokens-dir", default="/var/lib/letsencrypt/revocation_tokens/",
+        help="Directory where all revocation tokens are saved.")
+    add("--key-dir", default="/etc/letsencrypt/keys/", help="Keys storage.")
+    add("--cert-dir", default="/etc/letsencrypt/certs/",
+        help="Certificates storage.")
+
+    add("--le-vhost-ext", default="-le-ssl.conf",
+        help="SSL vhost configuration extension.")
+    add("--cert-path", default="/etc/letsencrypt/certs/cert-letsencrypt.pem",
+        help="Let's Encrypt certificate file.")
+    add("--chain-path", default="/etc/letsencrypt/certs/chain-letsencrypt.pem",
+        help="Let's Encrypt chain file.")
+
+    add("--apache-ctl", default="apache2ctl",
```

Comment:
I think we could easily fix this using https://docs.python.org/2/library/argparse.html#argument-groups and https://docs.python.org/2/library/argparse.html#formatter-class. Since this PR is already big, I would recommend to do it in a follow-up one.